### PR TITLE
Reexports

### PR DESCRIPTION
**Pull request by juancarlospaco**
- https://discord.com/channels/371759389889003530/768367394547957761/1119711750820986990
- https://github.com/nim-lang/Nim/pull/21812#issue-1701575795

### CODE REVIEW

**Comment by Araq** [comment]
Instead the docs should be built with `mm:orc` or something like that.

**Comment by juancarlospaco** [comment]
@Araq I dot not really understand what do you mean?, Im just trying to help,
if exported the docs go along with its documentation obviously, 
this PR makes focus on making the thing easily importable.

File: lib/system.nim [hunk 1]
```diff
@@ -2810,6 +2810,7 @@ proc arrayWith*[T](y: T, size: static int): array[size, T] {.raises: [].} =
       result[i] = `=dup`(y)
 
 
-when defined(nimV2) and hasThreadSupport:
-  from system/arc import isUniqueRef
-  export isUniqueRef
+when defined(nimV2):
+  when notJSnotNims and hasThreadSupport and hasAlloc and arcLikeMem and not declared(isUniqueRef):
+    from system/arc import isUniqueRef
```
Comment (metagn) [inline lib/system.nim:2815]: We have `when defined(nimV2): include system/arc` above, so `isUniqueRef` is already exported, and `not declared(isUniqueRef)` is always false, so this block doesn't do anything.

Comment (juancarlospaco) [reply lib/system.nim:2815]: Yes I seen it, but isUniqueRef is not declared/working in system devel.

Comment (metagn) [reply lib/system.nim:2815]: Yes it is?

```
> .\koch temp --lib:lib --warning[BareExcept]:off r --eval:"echo isUniqueRef (ref Exception)()"
...
true
```

**Pull request closed without merging**